src/interpreter/mod.rs

```diff
@@ -173,10 +175,12 @@ impl<'txin> Interpreter<'txin> {
     ///
     /// In case the script is actually dissatisfied, this may return several values
     /// before ultimately returning an error.
-    pub fn iter<'iter, F: FnMut(&KeySigPair) -> bool>(
+    /// See [Self::iter_assume_sigs] for a simpler API without information about Prevouts
+    /// but skips the signature verification
+    pub fn iter<'iter>(
         &'iter self,
-        verify_sig: F,
-    ) -> Iter<'txin, 'iter, F> {
+        verify_sig: Box<dyn FnMut(&KeySigPair) -> bool + 'iter>,
```

Comment:
i think you can make F: dyn FnMut... + 'iter without forcing it to be Boxed explicitly